src/read.cpp

```diff
@@ -46,6 +46,9 @@ CharacterVector read_lines_(List sourceSpec, List locale_,
 Function R6method(Environment env, const std::string& method) {
   return as<Function>(env[method]);
 }
+bool isTrue(SEXP x) {
+  return TYPEOF(x) == LGLSXP && Rf_length(x) == 1 && LOGICAL(x)[0] == TRUE;
```

Comment:
Maybe, our current R6 class tests with [`identical()`](https://github.com/hadley/readr/blob/a714a8ed01c6e8d51dd9bb790db0b6fbc2bf8146/R/callback.R#L70) so the value of `continue()` will always be either `TRUE` or `FALSE`, but I agree it makes sense to be explicit for user defined callback clasess.